src/App.js

```diff
@@ -8,46 +8,101 @@ import AddTodoForm from './addTodoForm';
-
-  useEffect(() =>{
-     new Promise((resolve,reject)=>{
-      setTimeout(() => {
-        resolve({ data: { todoList: initialList } });
-      }, 2000);
-     })
-     .then(result =>{
-       setTodoList(result.data.todoList)
-       setIsLoading(false)
-     })
-  }, [])
+  // let initialList = JSON.parse(localStorage.getItem('savedTodoList')) ?? []
+  const [isloading, setIsLoading] = useState(true)
 
+  const fetchData = async () => {
+    const options = {
+      method: 'GET',
+      headers: { 'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}` }
+    }
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`
+    try {
+      const response = await fetch(url, options)
 
-  useEffect(() => {
-    if (!isloading) {
-      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+      const data = await response.json();
+      const todos = data.records.map((todo) => {
+        return { title: todo.fields.Title, id: todo.id, createdAt: todo.createdTime }
+      })
+      const newTodos = todos.sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt))
+      setTodoList(newTodos)
+      setIsLoading(false)
+
+    } catch (error) {
+      console.log(error.message)
     }
-  }, [todoList])
+  }
+
+  const addTodo = async (url, data) => {
+    try {
+      const response = await fetch(url,
+        {
+          method: 'POST',
+          headers: {
+            'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({ records: [{ fields: data }] })
+        });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      const newData = await response.json();
+      const listItem = newData.records[0]
+      setTodoList(prevItems => [...prevItems, {
+        title: listItem.fields.Title,
+        id: listItem.id,
 
-  
-  function addTodo(newTodo) {
-    setTodoList(prevItems => [...prevItems, newTodo])
+      }])
+    } catch (error) {
+      console.error('Error updating todo:', error.message);
+      return null
+    }
   }
 
-  function removeTodo(id){
-    const newList =  todoList.filter(item=>
+  const removeTodo = async (id) => {
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`
+    try {
+      const response = await fetch(`${url}/${id}`, {
+        method: 'DELETE',
+        headers: {
+          'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        },
+      });
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      const newList = todoList.filter(item =>
         item.id !== id
       )
       setTodoList(newList)
```

Comment:
Nitpick:
You can leverage the state setter callback function to avoid creating another variable:
```diff
- const newList = todoList.filter(item =>
-   item.id !== id
- )
- setTodoList(newList)
+ setTodoList(prevTodoList => prevTodoList.filter(item => item.id !== id))
```

Reply:
Definitely. Thanks